agent/engine/task_manager.go

```diff
@@ -439,6 +448,39 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		mtask.Arn, container.Name, mtask.GetDesiredStatus().String())
 }
 
+// handleResourceChange updates a resource's known status
+func (mtask *managedTask) handleResourceChange(res taskresource.TaskResource,
```

Comment:
`updateResourceKnownStatus()` seems like a better name for this based on the description.

Reply:
it's more like attempting to update resource's known status. will update the description.